### DISCUSSION TITLE
ValuError for SPI calculation

### DISCUSSION THREAD

**Discussion by 08maria09** [monocongo/climate_indices#492] [Q&A]
Hi,

I came across an issue when trying to run an example command line invocation for daily SPI calculation. I used the example input file cmorph_lowres_daily_conus_prcp.nc. I get the following error message when I try to run the calculation.

![image](https://user-images.githubusercontent.com/121353952/209445130-b52b542a-fbc8-4229-8859-c74e7f30f97e.png)

The same error occured for my own files when I tried to calculate monthly SPI. I would be very grateful for any advice.

**Comment by monocongo** [comment]
Thanks for bringing this to our attention @08maria09 !

This has been fixed with #493 

Please try again with the latest from the master branch to confirm that you are OK now, if not please post an issue with error messages like the above. Thanks again for your help!

**Reply by 08maria09** [reply]
Sorry, I realised that I deleted my comment right after I saw your answer. I simply meant to add, that the same problem occured when I tried to calcuate monthly SPI with the example input file nclimgrid_lowres_prcp.nc. I also used some CMIP5 input files with which there was a problem with the order of the data's dimensions, but in that case there was an error message telling me what the problem was. But this time there was no error message and everything seemed to run smoothly until I looked at the results.

Once again thank you for your fast response over the holidays! If you manage to find a solution for the issue I would be very grateful for an update.

**Reply by monocongo** [reply]
I'm looking into this but no resolution yet. My assumption is that this is related to the latest version which removed the previous step involving NCO whereby the dimensions were properly re-oriented if not in (lat, lon, time) order. I may try to replace that functionality with [`xarray.DataArray.transpose()`](https://docs.xarray.dev/en/latest/generated/xarray.DataArray.transpose.html?highlight=transpose) but I'm not sure if it actually re-orients the data in memory so as to get the performance lift which that optimization gives.

**Comment by monocongo** [comment]
I'm unable to reproduce the error on my machine using the latest master branch. For example, here is the command I've run with log messages:
```
/home/james/miniconda3/envs/ci492/bin/python /home/james/git/climate_indices/src/climate_indices/__main__.py \
--index spi \
--periodicity daily \
--netcdf_precip /home/james/git/example_climate_indices/example/input/cmorph_lowres_daily_conus_prcp.nc \
--var_name_precip prcp \
--output_file_base /home/james/tmp/ci492/cmorph_lowres_daily_conus \
--scales 30 90 \
--calibration_start_year 1998 \
--calibration_end_year 2008 \
--multiprocessing all 
2023-01-04  13:46:25 INFO Start time:    2023-01-04 13:46:25.543124
2023-01-04  13:46:25 INFO Computing 30-day SPI/Pearson
2023-01-04  13:46:49 INFO Computing 30-day SPI/Gamma
2023-01-04  13:46:59 INFO Computing 90-day SPI/Pearson
2023-01-04  13:47:17 INFO Computing 90-day SPI/Gamma
2023-01-04  13:47:28 INFO End time:      2023-01-04 13:47:28.018201
2023-01-04  13:47:28 INFO Elapsed time:  0:01:02.475077

Process finished with exit code 0
```

The resulting datasets look reasonable. For example, here's a plot generated using Panoply: 
![cmorph_lowres_daily_conus_spi_pearson_30](https://user-images.githubusercontent.com/1328158/210628692-bcf2dd49-b9d7-4aae-8c4b-2e8103f32796.png)

**Comment by junniPan342** [comment]
I also met the same problem when I used the CHIRPS data. However, when the command starts with "spi" instead of "process_climate_indices --index spi", it can produce at least the results from the Gamma distribution.

![Screenshot 2023-06-15 173427](https://github.com/monocongo/climate_indices/assets/136643804/573ecb56-7cc4-46c5-b809-4477c42cd59d)
